File: src/struct_frame/boilerplate/py/payload_types/payload_minimal.py
# Payload Minimal
# Format: [MSG_ID] [PACKET]
# No length field, no CRC. Requires known message sizes.

from .base import PayloadType, PayloadConfig

PAYLOAD_MINIMAL_CONFIG = PayloadConfig(
    payload_type=PayloadType.MINIMAL,
    name="Minimal",
    has_crc=False,
    crc_bytes=0,
    has_length=False,
    length_bytes=0,
    has_sequence=False,
    has_system_id=False,
    has_component_id=False,
    has_package_id=False,
    description="[MSG_ID] [PACKET] - No length, no CRC. Requires known message sizes."
)

File: src/struct_frame/boilerplate/js/payload_types/payload_minimal.js
// Payload Minimal (JavaScript)

const { PayloadType } = require('./base');

const PAYLOAD_MINIMAL_CONFIG = {
    payloadType: PayloadType.MINIMAL,
    name: 'Minimal',
    hasCrc: false,
    crcBytes: 0,
    hasLength: false,
    lengthBytes: 0,
    hasSequence: false,
    hasSystemId: false,
    hasComponentId: false,
    hasPackageId: false,
    description: '[MSG_ID] [PACKET] - Minimal format with no length field and no CRC.'
};

module.exports = {
    PAYLOAD_MINIMAL_CONFIG
};
